### PR TITLE
Release 2023.2.1

### LINKED ISSUE #214: Set `pandas >=1.3` and `numpy >=1.21`
In https://github.com/dask/dask/pull/9950 we bumped our minimum versions `numpy` and `pandas` to `pandas >=1.3` and `numpy >=1.21`. This is just a reminder to do the same here when we push out the next release. 

cc @jakirkham

### PR DESCRIPTION
**Pull request by charlesbluca**
Closes #214

Checklist
* [ ] Used a [personal fork of the feedstock to propose changes](https://conda-forge.org/docs/maintainer/updating_pkgs.html#forking-and-pull-requests)
* [ ] Bumped the build number (if the version is unchanged)
* [ ] Reset the build number to `0` (if the version changed)
* [ ] [Re-rendered]( https://conda-forge.org/docs/maintainer/updating_pkgs.html#rerendering-feedstocks ) with the latest `conda-smithy` (Use the phrase <code>@<space/>conda-forge-admin, please rerender</code> in a comment in this PR for automated rerendering)
* [ ] Ensured the license file is being packaged.

### CODE REVIEW

**Comment by charlesbluca** [comment]
@conda-forge-admin, please rerender

File: recipe/meta.yaml [hunk 1]
```diff
@@ -24,10 +24,10 @@ requirements:
     - {{ pin_compatible('distributed', max_pin='x.x.x') }}
     - cytoolz >=0.8.2
     - lz4
-    - numpy >=1.18
-    - pandas >=1.0
+    - numpy >=1.21
+    - pandas >=1.3
     - bokeh >=2.4.2,<3
-    - jinja2
+    - jinja2 >=2.10.3
```
Comment (charlesbluca) [inline recipe/meta.yaml:30]: Seems like this was a change made with https://github.com/dask/distributed/pull/7285 that we missed in the 2023.1.0 release? cc @jrbourbeau @jakirkham just making sure that leaving this unconstrained wasn't intentional in the past few releases.

Shouldn't be difficult to submit a repodata patch to update this for the impacted versions

Comment (jakirkham) [reply recipe/meta.yaml:30]: Good catch! Should we update [the `jinja` optional requirement in `dask`]( https://github.com/dask/dask/blob/46c47be15661dd885e32d81f34d6d2bf114d52c8/setup.py#L21 ) as well?

Comment (charlesbluca) [reply recipe/meta.yaml:30]: Yeah good catch, bumping that makes sense to me

Comment (charlesbluca) [reply recipe/meta.yaml:30]: Opened up https://github.com/dask/dask/pull/9999 to handle this

**Comment by jakirkham** [comment]
@conda-forge-admin, please restart CI

**Pull request merged**